lib/truemail/validate/resolver_execution_wrapper.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+module Truemail
+  module Validate
+    class ResolverExecutionWrapper
+      class << self
+        attr_writer :attempts
+
+        def attempts
```

Comment:
I think it should be a class instance method.
In your case attempts value will be shared between all each ResolverExecutionWrapper call